base/applications/osk/main.c

```diff
@@ -585,12 +585,35 @@ int WINAPI wWinMain(HINSTANCE hInstance,
                     int show)
 {
     HANDLE hMutex;
+    DWORD dwError;
     INT LayoutResource;
 
     UNREFERENCED_PARAMETER(prev);
     UNREFERENCED_PARAMETER(cmdline);
     UNREFERENCED_PARAMETER(show);
 
+    /*
+        Obtain a mutex for the program. This will ensure that
+        the program is launched only once.
+    */
+    hMutex = CreateMutexW(NULL, FALSE, L"OSKRunning");
+
+    if (hMutex)
+    {
+        /* Check if there's already a mutex for the program */
+        dwError = GetLastError();
+
+        if (dwError == ERROR_ALREADY_EXISTS)
+        {
+            /*
+                A mutex with the object name has been created previously.
+                To avoid any occurence of race conditions, do not create a new mutex.
```

Comment:
This line doesn't make much sense to me. You wouldn't be able to create a different mutex with the same name even if you wanted to, and there's no race condition unless you do the OpenMutex/CreateMutex dance that the code previously did.
Perhaps just `Therefore, another instance is already running.` or similar?